### DISCUSSION TITLE
Limit zoom to keep the network viz within the square area actually rendered

### DISCUSSION THREAD

**Discussion by Javingka** [cosmograph-org/cosmos#66] [Q&A]
Hi everyone.

Fist of all, I have to say you have build an incredibly good tool for network visualization, simple and powerful. It gives an easy and seamless  interface to a rather complex endeavor, is not just webGL usage it is also network visualization, so congrats!. 

Now, I'm having an issue I hope is easy to solve. I want to know how would you fix the zoom ratio to avoid go outside the vistual square canvas used.

You can see in the image if I zoom out too much I can see where the square edge is. I have no problem in have all my nodes squeezed on the edge, but I do have problem letting that edge visible.

I hope my doubt is clear. And thanks so much in advance for your help 
 
![Captura de Tela 2023-06-14 às 11 53 26](https://github.com/cosmograph-org/cosmos/assets/2746141/f46e639a-c6e2-4090-affb-22de96adfe45)

regards.

**Comment by rokotyan** [comment]
@Javingka Thanks for the kind words! Will you be able to share your data and configuration, probably over StackBlitz or a similar tool? That will help us to debug the issue.
